mono/mini/mini.c

```diff
@@ -3577,6 +3593,11 @@ mini_method_compile (MonoMethod *method, guint32 opts, MonoDomain *domain, JitFl
 		mono_cfg_dump_ir (cfg, "local_cprop");
 	}
 
+	MONO_TIME_TRACK (mono_jit_stats.jit_decompose_typechecks, mono_decompose_typechecks (cfg));
```

Comment:
We're inlining after the front end and that will change the CFG. Good thing you picked a place before the CFG passes and analysis. We do this before SSA formation, which is ok too.

But, said that, what's the reasoning to have it here? We're losing a lot of optimizations that could take advantage of longer BBs.

We could do it all the way down after DCE and AA. Could this pass introduce issues?

Given DCE is crazy cheap, we can do another DCE after decompose.